src/coreclr/src/vm/tieredcompilation.cpp

```diff
@@ -448,8 +448,7 @@ void TieredCompilationManager::DeactivateTieringDelay()
         COUNT_T methodCount = methodsPendingCounting->GetCount();
         CodeVersionManager *codeVersionManager = GetAppDomain()->GetCodeVersionManager();
 
-        MethodDescBackpatchInfoTracker::PollForDebuggerSuspension();
-        MethodDescBackpatchInfoTracker::ConditionalLockHolder slotBackpatchLockHolder;
+        MethodDescBackpatchInfoTracker::ConditionalLockHolderForGCCoop slotBackpatchLockHolder;
```

Comment:
why is tiering enabled while under the debugger? Wouldnt the debugger enforce non-optimized code anyways?

Reply:
I'm not sure what the relevance is to the code pointed out, can you clarify? Aside from that when tiering is enabled under the debugee (typical case), then the debugger needs to be able to handle that. Methods jitted while the debugger is attached may be generated with unoptimized code, but methods that are already jitted with optimizations when a debugger is attached are still optimized.

Reply:
Not relevant to the specific change, just commented since it was in tieredcompilation.cpp. Was just wondering why not just disable tiering when under the debugger.. is there a use case where Tiering is helpful while debugging?

Reply:
When a debugger attaches to an already-running process, it is typically attaching to a process that has tiering enabled, at an arbitrary point in time. The debugger needs to be able to handle the effects of tiering in the debugee despite any desire to disable or defer tiering activities while a debugger is attached. As for that desire, whether it is necessary or useful is probably a separate question.